framework/include/problems/FEProblemBase.h

```diff
@@ -489,6 +491,24 @@ class FEProblemBase : public SubProblem, public Restartable
   virtual bool hasFunction(const std::string & name, THREAD_ID tid = 0);
   virtual Function & getFunction(const std::string & name, THREAD_ID tid = 0);
 
+  /**
+   * add a MOOSE line search
+   */
+  virtual void addLineSearch(const InputParameters & /*parameters*/)
+  {
+    mooseError("Line search not implemented for this problem type yet.");
+  }
+
+  /**
+   * execute MOOSE line search
+   */
+  virtual void linesearch();
```

Comment:
Casing is incorrect here and throughout all methods that override. Let's get this one changed before this goes in since you use `lineSearch` in several other places.